hbase-server/src/test/java/org/apache/hadoop/hbase/regionserver/TestRegionServerHostname.java

```diff
@@ -205,4 +204,22 @@ public void testRegionServerHostnameReportedToMaster() throws Exception {
       assertEquals(expectedRS, servers.size());
     }
   }
+
+  private boolean ignoreNetworkInterface(NetworkInterface networkInterface) throws Exception {
+    return networkInterface == null || networkInterface.isLoopback() || networkInterface.isVirtual()
+      || !networkInterface.isUp();
+  }
+
+  private List<NetworkInterface> getValidNetworkInterfaces() throws Exception {
+    List<NetworkInterface> validNetworkInterfaces = new LinkedList<>();
```

Comment:
Just use array list, only in very very rare cases LinkedList could have better performance...